docs/API.md

```diff
@@ -559,6 +560,58 @@ if err != nil {
 }
 ```
 
+<a name="ComposeObject"></a>
+### ComposeObject(dst DestinationInfo, srcs []SourceInfo) error
```

Comment:
`NewSSEInfo`, `NewSourceInfo` and `NewDestinationInfo` are added in docs. The types are only documented via godoc for now, in keeping with the existing style (e.g. CopyCondition was documented only via godoc). We may add documentation for types in the API.md later but that is a matter for discussion.